madmom/features/beats.py

```diff
@@ -1238,6 +1238,380 @@ def add_arguments(parser, min_bpm=MIN_BPM, max_bpm=MAX_BPM,
         return g
 
 
+class MultiAgentBeatTrackingProcessor(OnlineProcessor):
+    """
+    Beat Tracking via Multiple Agents. Oriented towards the paper "Beat
+    Tracking for multiple applications: A multi-agent system architecture with
+    state recovery" by Lobate Oliveira et al., 2012
+
+    Parameters
+    ----------
+    max_agents : int
+        Max number of agents.
+    tempi : int
+        Number of tempi to be considered.
+    induction_time : float
+        Window length in seconds used for inducting the agents.
+
+    """
+    MAX_AGENTS = 30
+    TEMPI = 3
+    INDUCTION_TIME = 2.
+
+    class Agent(object):
+        """
+        Agent class for tracking the beats. Each agent has a tempo, a score,
+        a current prediction where the next beat is and a complete history
+        of all previously detected beats.
+
+        Parameters
+        ----------
+        inner_window : int
+            Inner window in frames to each side where a beat is accepted.
+        outer_window : float
+            outer window factor to each side depending on the interval.
+        threshold : float
+            Threshold value for accepting a beat.
+        correction_factor : float
+            Allow agent to adapt to errors.
+        inherit_score_factor : float
+            Child agents will inherit a percentage of their parents.
+
+        """
+        # TODO: Should we use @classmethod to set all values from outside?
```

Comment:
@classmethod is probably not the right choice for this. Usually these parameters are just added to `__init__`. 

Since you documented them here (as numpydoc requires the init-params to be documented at class-level), these should be available to `__init__`. Vice versa, the params of init must be documented here as well.